pkg/security/registry/podsecuritypolicysubjectreview/rest.go

```
@@ -0,0 +1,115 @@
+
+// New creates a new PodSecurityPolicySubjectReview object
+func (r *REST) New() runtime.Object {
+	return &securityapi.PodSecurityPolicySubjectReview{}
+}
+
+// Create registers a given new PodSecurityPolicySubjectReview instance to r.registry.
+func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
+	pspsr, ok := obj.(*securityapi.PodSecurityPolicySubjectReview)
+	if !ok {
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("not a PodSecurityPolicySubjectReview: %#v", obj))
+	}
+
+	ns, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, kapierrors.NewBadRequest("namespace parameter required.")
+	}
+
+	if errs := securityvalidation.ValidatePodSecurityPolicySubjectReview(pspsr); len(errs) > 0 {
+		return nil, kapierrors.NewInvalid(securityapi.Kind(pspsr.Kind), "", errs)
+	}
+
+	userInfo := &user.DefaultInfo{Name: pspsr.Spec.User, Groups: pspsr.Spec.Groups}
+	matchedConstraints, err := r.sccMatcher.FindApplicableSCCs(userInfo)
+	if err != nil {
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("unable to find SecurityContextConstraints: %v", err))
+	}
+
+	saName := pspsr.Spec.Template.Spec.ServiceAccountName
+	if len(saName) > 0 {
+		saUserInfo := serviceaccount.UserInfo(ns, saName, "")
+		saConstraints, err := r.sccMatcher.FindApplicableSCCs(saUserInfo)
+		if err != nil {
+			return nil, kapierrors.NewBadRequest(fmt.Sprintf("unable to find SecurityContextConstraints: %v", err))
+		}
+		matchedConstraints = append(matchedConstraints, saConstraints...)
+	}
+	oscc.DeduplicateSecurityContextConstraints(matchedConstraints)
+	sort.Sort(oscc.ByPriority(matchedConstraints))
+	var namespace *kapi.Namespace
+	for _, constraint := range matchedConstraints {
+		var (
+			provider kscc.SecurityContextConstraintsProvider
+			err      error
+		)
+		if provider, namespace, err = oscc.CreateProviderFromConstraint(ns, namespace, constraint, r.client); err != nil {
+			glog.Errorf("Unable to create provider for constraint: %v", err)
+			continue
```

Again, swallowing errors.